src/runner.ts

```diff
@@ -1,17 +1,163 @@
 import * as core from "@actions/core";
 import * as types from './types';
+import * as cp from 'child_process';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as pt from 'path';
+import * as glob from 'glob';
+import {messages, messagesFormatter} from './messages';
+
+export interface RunOptions {
+    /* Specify a path to folder or a file for the Parasoft coverage report */
+    report: string;
+
+    /* Specify a path to Parasoft tools installation folder or Java installation folder */
+    parasoftToolOrJavaRootPath: string;
+}
 
 export interface RunDetails {
     exitCode: number;
+    convertedCoberturaReportPaths?: string[];
 }
 
 export class CoverageParserRunner {
-    async run() : Promise<RunDetails> {
-        // TODO: Simulate coverageNode input for testing the structure implemented in current task
-        const coberturaCoverage: types.CoberturaCoverage = this.getCoberturaCoverage();
+    async run(runOptions: RunOptions) : Promise<RunDetails> {
+        const parasoftReportPaths = this.findParasoftCoverageReports(runOptions.report);
+        if (!parasoftReportPaths || parasoftReportPaths.length == 0) {
+            return Promise.reject(messagesFormatter.format(messages.coverage_report_not_found, runOptions.report));
+        }
+
+        const javaFilePath = this.getJavaFilePath(runOptions.parasoftToolOrJavaRootPath);
+        if (!javaFilePath) {
+            return { exitCode: -1 }
+        }
+
+        const outcome = await this.convertReportsWithJava(javaFilePath, parasoftReportPaths);
+        if (outcome.exitCode == 0 && outcome.convertedCoberturaReportPaths) {
+            // TODO: Implement Merge multiple cobertura reports
+            // TODO: Implement calculating coverage data from the converted report
+            // TODO: Simulate coverageNode input for testing the structure implemented in current task
+            const coberturaCoverage: types.CoberturaCoverage = this.getCoberturaCoverage();
+
+            await this.generateCoverageSummary(coberturaCoverage);
+        }
+
+        return { exitCode: outcome.exitCode };
+    }
+
+    private findParasoftCoverageReports(reportPath: string) : string[] | undefined {
```

Comment:
Actually, I have a design-related question here. I guess the reason we allow users to specify either a directory or a file here is to align with SOAtest's GitHub Action. However, in the SOAtest Action, we actually handle only a single test report. 
On other platforms that support coverage, we use a matching pattern approach. For example, in Azure, we use minimatch. I think we might want to align with other platforms that display coverage.